src/pages/Create.tsx

```diff
@@ -1,174 +1,175 @@
+import ReactQuill from 'react-quill';
+import 'react-quill/dist/quill.snow.css';
+import Header from '../components/Header';
+import ImageOutlinedIcon from '@mui/icons-material/ImageOutlined';
+import EditIcon from '@mui/icons-material/Edit';
+import {
+    ErrorButton,
+    OutlinedButton,
+    PrimaryButton
+} from '../components/Button';
+import {useAppContext} from '../context/AppContext';
+import {Blog} from '../@types/interfaces';
+import {BlogContract} from '../@types/enums';
+import PageLayout from '../layouts/PageLayout';
+import {AddBlogContractParams, EditBlogContractParams} from '../@types/types';
+
+const Create = ({edit}: { edit?: boolean }) => {
+    const navigate = useNavigate();
+    const {ownerAddress, blogs, getAllBlogs} = useAppContext();
+
+    const [editId, setEditId] = useState<number | undefined>();
+    const [loading, setLoading] = useState<boolean>(false);
+    const [cover, setCover] = useState<string | ArrayBuffer | null>('');
+    const [title, setTitle] = useState<string>('');
+    const [content, setContent] = useState<string>('');
+
+    useEffect(() => {
+        const id = window.location.search.slice(4);
+        if (edit && !blogs.loading) {
+            const reqBlog = blogs.data.find((blog) => blog.storageHash === id);
+            setEditId(reqBlog!.id);
+            setCover(reqBlog!.coverImage);
+            setTitle(reqBlog!.title);
+            setContent(reqBlog!.content);
+        }
+    }, [blogs, edit]);
+
+    const handleFileInput = (e: any) => {
+        const reader = new FileReader();
+        reader.onload = function (e: any) {
+            setCover(e.target.result);
+        };
+        reader.readAsDataURL(e.target.files[0]);
+    };
+
+    const handleSave = async (isPublished: boolean) => {
+        setLoading(true);
+        const now = dayjs().format('MMM DD, YYYY');
+
+        const form = JSON.stringify({
+            coverImage: cover,
+            title,
+            content,
+            publisher: ownerAddress,
+            createdDate: now
+        } as Blog);
+        const file = new File([form], 'blog.json', {type: 'application/json'});
+
+        const formData = new FormData();
+        formData.append('file', file);
+        // Upload the File to arweave
+        const res = await window.point.storage.postFile(formData);
+        setLoading(false);
+        // Save data to smart contract
+        if (edit) {
+            await window.point.contract.send({
+                contract: BlogContract.name,
+                method: BlogContract.editBlog,
+                params: [editId, res.data, now] as EditBlogContractParams
+            });
+        } else {
+            await window.point.contract.send({
+                contract: BlogContract.name,
+                method: BlogContract.addBlog,
+                params: [res.data, isPublished, now] as AddBlogContractParams
+            });
+        }
+        getAllBlogs();
+        navigate('/');
```

Comment:
@JatinBumbra FYI `navigate(-1)` was breaking the browser :) I don't know, why, it's a documented feature.